ssl/statem/extensions_clnt.c

```diff
@@ -199,11 +200,12 @@ EXT_RETURN tls_construct_ctos_supported_groups(SSL_CONNECTION *s, WPACKET *pkt,
     }
 
     /*
-     * We only support EC groups in TLSv1.2 or below, and in DTLS. Therefore
+     * We only support EC groups in (D)TLSv1.2 or below, and in DTLS. Therefore
      * if we don't have EC support then we don't send this extension.
      */
     if (!use_ecc(s, min_version, max_version)
-            && (SSL_CONNECTION_IS_DTLS(s) || max_version < TLS1_3_VERSION))
+           && ((!isdtls && max_version < TLS1_3_VERSION)
+           || (isdtls && DTLS_VERSION_LT(max_version, DTLS1_3_VERSION))))
```

Comment:
I would line up the || with the second ( on the line above.

Or alternatively we could break this up into more than one "if" since its getting quite complicated.